src/events/types/inventory/EnchantmentsToAdd.ts

```diff
@@ -0,0 +1,74 @@
+import { Block } from '../../../block/Block';
+import { ItemStack } from '../../../material/ItemStack';
+import { Player } from '../../../player/Player';
+import { Cancellable } from '../../Cancellable';
+import { InventoryEvent } from './InventoryEvent';
+
+/**
+ * Called when an ItemStack is successfully enchanted (currently at enchantment table)
+ */
+export class EnchantItemEvent extends InventoryEvent implements Cancellable {
+
+  public static getBukkitClasspath(): string {
+    return 'org.bukkit.event.enchantment.EnchantItemEvent';
+  }
+
+  /**
+   * Gets the block being used to enchant the item
+   */
+  public getEnchantBlock(): Block {
+    return Block.fromJava(this.toJava().getEnchantBlock());
+  }
+
+  /**
+   * Gets the player enchanting the item
+   */
+  public getEnchanter(): Player {
+    return Player.fromJava(this.toJava().getEnchanter());
+  }
+
+  // TODO: implement
+  // /**
+  //  * Get map of enchantment (levels, keyed by type) to be added to item (modify map returned to change values). Note: Any enchantments not allowed for the item will be ignored
+  //  */
+  // public getEnchantsToAdd(): Map<Enchantment, number> {
+  //   return this.toJava().getEnchantsToAdd();
+  // }
```

Comment:
@connerdouglass I do need some help on this one

Reply:
I think we should skip this method for this PR, and then open an Issue to implement it separately.